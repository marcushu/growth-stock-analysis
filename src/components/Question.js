import React from 'react';
import PropTypes from 'prop-types';

import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col'


class Question extends React.Component {
  constructor(props) {
    super(props);

    let scoreSeed = this.props.selected === "false" ?
      this.props.falseValue :
      this.props.trueValue;

    this.state = {
      currentScore: scoreSeed,
    }

    Question.propTypes = {
      description: PropTypes.string,
      selected: PropTypes.string,
      trueValue: PropTypes.number,
      falseValue: PropTypes.number,
      changeHandler: PropTypes.func
    }
  }

  updateValue = e => {
    this.setState({ selected: e.target.value })

    if (e.target.value === 'true') {
      this.props.changeHandler({
        newVal: this.props.trueValue,
        oldVal: this.state.currentScore,
        description: this.props.description,
        selected: e.target.value
      });
      this.setState({ currentScore: this.props.trueValue });

    } else {
      this.props.changeHandler({
        newVal: this.props.falseValue,
        oldVal: this.state.currentScore,
        description: this.props.description,
        selected: e.target.value
      });
      this.setState({ currentScore: this.props.falseValue });

    }
  }

  render() {
    return (
      <>
        <Row>
          <Col xs={9}>
            {this.props.description}
          </Col>
          <Col>
            <Form >
              <span key={`inline-${'radio'}`} className="mb-3">
                <Form.Check
                  inline
                  type="radio"
                  label="Yes"
                  name="formHorizontalRadios"
                  id="formHorizontalRadios1"
                  onChange={this.updateValue}
                  value={"true"} // this can only be number || string
                  checked={this.props.selected === "true"}
                />
                <Form.Check
                  inline
                  type="radio"
                  label="No"
                  name="formHorizontalRadios"
                  id="formHorizontalRadios2"
                  onChange={this.updateValue}
                  value={"false"}
                  checked={this.props.selected === "false"}
                />
              </span >
            </Form>
          </Col>
        </Row>
      </>
    )
  }

}

export default Question

