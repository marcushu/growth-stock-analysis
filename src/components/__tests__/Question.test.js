import React from 'react';
import { render,  cleanup } from '@testing-library/react';
import Question from '../Question';

afterEach(cleanup);

test("renders ok", () => {
  const { asFragment } = render(
    <Question
    description={'test_description'}
    trueValue={1}
    falseValue={-1}
    changeHandler={() => {}}
    selected={'false'} />
  );

  expect(asFragment()).toMatchSnapshot();
});

test('renders the label', () => {
  const { getByText } = render(
    <Question
    description={'test_description'}
    trueValue={1}
    falseValue={-1}
    changeHandler={() => {}}
    selected={'false'} />
  );

    expect(getByText('test_description')).toBeInTheDocument();
})