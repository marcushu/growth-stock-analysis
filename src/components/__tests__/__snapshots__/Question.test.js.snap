// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-9"
    >
      test_description
    </div>
    <div
      class="col"
    >
      <form
        class=""
      >
        <span
          class="mb-3"
        >
          <div
            class="form-check form-check-inline"
          >
            <input
              class="form-check-input"
              id="formHorizontalRadios1"
              name="formHorizontalRadios"
              type="radio"
              value="true"
            />
            <label
              class="form-check-label"
              for="formHorizontalRadios1"
              title=""
            >
              Yes
            </label>
          </div>
          <div
            class="form-check form-check-inline"
          >
            <input
              checked=""
              class="form-check-input"
              id="formHorizontalRadios2"
              name="formHorizontalRadios"
              type="radio"
              value="false"
            />
            <label
              class="form-check-label"
              for="formHorizontalRadios2"
              title=""
            >
              No
            </label>
          </div>
        </span>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
